shared/src/main/scala/org/adridadou/openlaw/parser/template/printers/XHtmlAgreementPrinter.scala

```diff
@@ -154,18 +153,35 @@ case class XHtmlAgreementPrinter(preview: Boolean, paragraphEdits: ParagraphEdit
         case section@SectionElement(_, level, _, _, _, _) =>
           // partition out all content that will be within the newly defined sections
           val higherLevel = level - 1
-          val (content, remaining) = partitionAt(xs) { case SectionElement(_, thisLevel, _, _, _, _) if thisLevel === higherLevel => true }
-
+          val (content, remaining) = partitionAt(xs) { 
+            case SectionElement(_, thisLevel, _, _, _, _) if thisLevel === higherLevel => true 
+          }
           // Partition the elements into sections at this level
           val sections = partitionSections(level, section +: content)
-
-          val frag = ul(`class` := s"list-lvl-$level")(
-            sections.map { section =>
-              recurse(section._2, conditionalBlockDepth, true, { elems => Seq(li(elems)) })
-            }
-          )
-          tailRecurse(remaining, conditionalBlockDepth, inSection, { elems => continue(frag +: elems) } )
-
+          var afterBreak:Seq[AgreementElement] = Seq()
```

Comment:
Instead of accumulating with a mutable variable here, I think the more idiomatic way to do this would be folding, e.g.

```
val afterBreak = sections.foldLeft(Seq[AgreementElement]()) { case (accu, (_, elements)) =>
  // bifurcate the sections around the section break
  val (_, after) = elements.span(_ != Paragraph(List(FreeText(SectionBreak))))
  accu ++ after
}
```